consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,30 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{ServiceNode, UnexpectedResponse}
+import io.buoyant.namer.Metadata
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  private[this] lazy val domainFuture: Future[String] =
```

Comment:
10s seems like a reasonable polling interval to me.  Just a warning that this is going to bit tricky to get right because `Var.async`s will stop updating if they don't have any listeners which will be true most of the time.  We probably don't want the `Var.async` to restart and fire a new request for every consul change.  (or maybe we do, in which case we can ditch `Var.async` and just make a Future call each time.)